cli/tests/program.rs

```diff
@@ -1614,3 +1825,176 @@ fn test_cli_program_dump() {
+    config.signers = vec![online_signer, buffer_signer_identity];
+    config.command = CliCommand::Program(ProgramCliCommand::Deploy {
+        program_location: None,
+        fee_payer_signer_index: 0,
+        program_signer_index: None,
+        buffer_signer_index: Some(1),
+        allow_excessive_balance: false,
+        upgrade_authority_signer_index: 0,
+        is_final: false,
+        max_len: None,
+        skip_fee_check: false,
+    });
+    config.output_format = OutputFormat::JsonCompact;
+    let error = process_command(config).unwrap_err();
+    assert_eq!(error.to_string(), "Deploying program failed: RPC response error -32002: Transaction simulation failed: Error processing Instruction 1: Incorrect authority provided [5 log messages]");
+}
+
+fn verify_deployed_program<'a>(
+    config: &mut CliConfig<'a>,
+    online_signer: &'a Keypair,
+    offline_signer_pub_key: Pubkey,
+    program_signer_pub_key: Pubkey,
+    max_program_data_len: usize,
+    min_slot: u64,
+    max_slot: u64,
+) {
+    // Verify show
+    config.signers = vec![online_signer];
+    config.command = CliCommand::Program(ProgramCliCommand::Show {
+        account_pubkey: Some(program_signer_pub_key),
+        authority_pubkey: program_signer_pub_key,
+        get_programs: false,
+        get_buffers: false,
+        all: false,
+        use_lamports_unit: false,
+    });
+    let response = process_command(config);
+    let json: Value = serde_json::from_str(&response.unwrap()).unwrap();
+    let address_str = json
+        .as_object()
+        .unwrap()
+        .get("programId")
+        .unwrap()
+        .as_str()
+        .unwrap();
+    assert_eq!(
+        program_signer_pub_key,
+        Pubkey::from_str(address_str).unwrap()
+    );
+    let programdata_address_str = json
+        .as_object()
+        .unwrap()
+        .get("programdataAddress")
+        .unwrap()
+        .as_str()
+        .unwrap();
+    let (programdata_pubkey, _) = Pubkey::find_program_address(
+        &[program_signer_pub_key.as_ref()],
+        &bpf_loader_upgradeable::id(),
+    );
+    assert_eq!(
+        programdata_pubkey,
+        Pubkey::from_str(programdata_address_str).unwrap()
+    );
+    let authority_str = json
+        .as_object()
+        .unwrap()
+        .get("authority")
+        .unwrap()
+        .as_str()
+        .unwrap();
+    assert_eq!(
+        offline_signer_pub_key,
+        Pubkey::from_str(authority_str).unwrap()
+    );
+    let deployed_slot = json
+        .as_object()
+        .unwrap()
+        .get("lastDeploySlot")
+        .unwrap()
+        .as_u64()
+        .unwrap();
+    assert!(deployed_slot >= min_slot);
+    assert!(deployed_slot <= max_slot);
+    let data_len = json
+        .as_object()
+        .unwrap()
+        .get("dataLen")
+        .unwrap()
+        .as_u64()
+        .unwrap();
+    assert_eq!(max_program_data_len, data_len as usize);
+}
```

Comment:
Rather than testing through the `Show` command, it would be much simpler and more complete to fetch the program data account and make sure that the data is expected, as in `test_cli_program_deploy_with_authority`

```
    let (programdata_pubkey, _) =
        Pubkey::find_program_address(&[program_pubkey.as_ref()], &bpf_loader_upgradeable::id());
    let programdata_account = rpc_client.get_account(&programdata_pubkey).unwrap();
    assert_eq!(
        programdata_account.data[UpgradeableLoaderState::size_of_programdata_metadata()..],
        program_data[..]
    );
```

Reply:
Hm, right, changed in https://github.com/solana-labs/solana/pull/33860/commits/1736f75b9753d6aca90e1d0ccfae87ec59b37d31

I've "copied" the Show-approach from another test, overall "checking same thing in different ways" results in better code-coverage - sometimes there is a merit to doing that (probably not here though).